### ISSUE TITLE
Roles should not inadvertently grant access

### ISSUE THREAD

**Issue by programmerq** [gravitational/teleport#5513]
### Description

**What happened**:

When creating a role that only specifies node labels, application and kubernetes access are fully granted.

Take this short example role:

```yaml
version: v3
kind: role
metadata:
  name: qa
spec:
  allow:
    node_labels:
      'account_type': 'QA'
```

When this is created via `tctl create -f qa.yaml`, it results in the following `tctl get roles/qa`:

```yaml
kind: role
metadata:
  id: 1612814108898420391
  name: qa
spec:
  allow:
    app_labels:
      '*': '*'
    kubernetes_labels:
      '*': '*'
    node_labels:
      account_type: QA
  deny: {}
  options:
    cert_format: standard
    enhanced_recording:
    - command
    - network
    forward_agent: false
    max_session_ttl: 30h0m0s
    port_forwarding: true
version: v3
```

Since there was no initial mention of the app nor kube services, this could lead to a situation where a user with this QA role will be allowed to access apps/kube clusters when it was only intended for her to get access to nodes with the `account_type: QA` label.

**What you expected to happen**:

Teleport should follow the concept of least privileged access. no access should be granted to kubernetes clusters nor applications.

**How to reproduce it (as minimally and precisely as possible)**:

Copy/paste the example yaml given above and run the commands above.

### Environment

Environment 1

- Teleport version (use `teleport version`): `Teleport Enterprise v5.0.0 git:v5.0.0-0-gac4971801 go1.15.5`
- Tsh version (use `tsh version`): `Teleport v5.0.0 git:v5.0.0-0-gac4971801 go1.15.5`
- OS (e.g. from `/etc/os-release`): Amazon Linux 2

Environment 2

- Teleport version (use `teleport version`): `Teleport Enterprise v5.1.2 git:v5.1.2-0-g822d10b44 go1.15.5`
- Tsh version (use `tsh version`): `Teleport v5.1.2 git:v5.1.2-0-g822d10b44 go1.15.5`
- OS (e.g. from `/etc/os-release`): quay.io/gravitational/teleport-ent:5.1.2 container image on EKS

- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware):

AWS

**Browser environment**

N/A

**Relevant Debug Logs If Applicable**

N/A

**Comment by klizhentas** [comment]
@russjones We have done this for backwards compatibility, but newly created roles should not grant access to kubernetes and app clusters by default unless specified.

**Comment by andrejtokarcik** [comment]
This is related to my questions in #5428. All the kinds of labels except for `cluster_labels` set an allow-all wildcard as their (overly optimistic) default.

**Comment by tcsc** [comment]
@klizhentas, @andrejtokarcik - So, If I'm reading this correctly, the solution is to apply the wildcard labels to roles migrated from older versions, but not to newly-created roles?

**Comment by nklaassen** [comment]
@andrejtokarcik Do we currently have a migration path where we would be able to grant the permissive privileges to admin roles created on older releases? Dropping the defaults within 6.x seems fine as all roles will be persisted with the wildcard privileges, but do we need a way to grant e.g. database access to admin roles created on 5.x?

Dropping the `cluster_labels` logic seems a bit tricky as well, we probably don't want a role created on 6.1 to lose cluster permissions on 6.2. We should probably start by adding the wildcard privileges to the built-in admin role, but I'm not sure when it would be safe to remove the current [cluster_labels](https://github.com/gravitational/teleport/blob/fbc2e4eafbc952f5a6c44b7707996f94f5683829/lib/services/role.go#L1326-L1344) logic.

**Comment by andrejtokarcik** [comment]
@nklaassen I guess these concerns could be addressed similarly to [how RBAC was introduced to OSS](https://github.com/gravitational/teleport/blob/fbc2e4eafbc952f5a6c44b7707996f94f5683829/lib/auth/init.go#L529-L534).

Regarding the admin role itself the situation could be simplified if the admin role was always simply overwritten with `NewAdminRole` [during auth server initialization](https://github.com/gravitational/teleport/blob/fbc2e4eafbc952f5a6c44b7707996f94f5683829/lib/auth/init.go#L301-L309).

**Comment by nklaassen** [comment]
@andrejtokarcik I didn't realize we overwrite the admin role during auth server init, that makes a lot of sense. That honestly alleviates my concerns above since there's no way an admin role could lose privileges on an upgrade.

**Comment by andrejtokarcik** [comment]
@nklaassen We currently do *not* overwrite the admin role (if the admin role already exists, `CreateRole` will return `AlreadyExists` that is then handled in the init snippet). I merely suggested we perhaps should force such an overwrite.

**Comment by nklaassen** [comment]
@andrejtokarcik Ahh you're right, I had tried it by upgrading from 5.2 to master but of course the OSS migration is what updated the admin role, not that init snippet. Honestly, I think the current migration should be enough, since 5.x admins will get db access there and there's no way for current roles to lose access to anything.

Actually this does make the cluster_labels situation tricky, since the current admin is not stored with those wildcards but would currently be granted access. I will file a separate issue for this.

**Comment by nklaassen** [comment]
The fix for this is the addition of V4 roles which have stricter defaults for labels, which merged to master in #7118 and backported to 6.2 in #7269. This should be released in Teleport 6.2.4
